### DISCUSSION TITLE
Netherlands: Vrijopnaam

### DISCUSSION THREAD

**Discussion by GeoffreyFrogeye** [JaccoR/hass-entso-e#90] [Price Modifyer Templates]
Template for the dynamic rates contract for Vrijopnaam.
VAT is handled by the template, set the plugin VAT option to 0.

You can subscribe to notifications as I'll announce if the template needs to be adjusted for upcoming changes in rates.
I periodically check myself, but feel free to comment too if you notice I'm late.
Last I checked (and bothered to update this message), the template is valid until at least: **April 2025**.

```jinja2
{% set inkoopvergoeding = 0.0165 %}
{% set energiebelasting = 0.10154 %}
{% set btw = 0.21 %}
{{ (current_price + inkoopvergoeding + energiebelasting) * (1 + btw) }}
```

(I don't follow the `s` dictionnary convention so it's easier to have templates that can adapt to scheduled rate changes.)

**Comment by cyborghound** [comment]
Does this still match with you sheets? I implemented your template but Entsoe show a much bigger price then what I see in Vrijopnaam.app.

**Reply by GeoffreyFrogeye** [reply]
In the settings (Mijn gegevens > Prijsweergave (dynamische prijzen)) you can set if the prices are all inclusive or not. I made the calculations to match the "All-in-prijs", do you have the same setting? It still matches for me. I just checked and the rates are the same since February 2023 to May 2023.

If yes, then I assume there are different rates depending on the location or something? Would you mind sharing the sheet of the current month (Contractgegevens > Tariefbladen) and the average price for yestarday (Stroom inzicht > Zondag 16 April 2023 > Prijs per uur - gemiddeld X)? Mine is "0,31713 €/kWh". So if it does match, maybe it's something with the app (did not leave the VAT field empty?).

**Reply by cyborghound** [reply]
Hi there, I checked but I also have All-In prices. An example for today is the price for 18:00-19:00 in Vrijopnaam.app is 0,33193 / kWh in the table under Vrijopnaam -> Tarieven. Home Assistant with this integration tells me a current price of 0.32043 €/kWh. I use 0 VAT in the integration itself and the script from above. I only changed the inkoopvergoeding because that's 0.0165 for me. My average for today is 0,33378. The strange thing is that when I change the script and save it, the sensors won't ever change. Looks like something wrong with the integration maby.

**Reply by GeoffreyFrogeye** [reply]
> I only changed the inkoopvergoeding because that's 0.0165 for me

Is it really? Because I also had a value of 0.33193 €/kWh for 18:00-19:00 in Vrijopnaam.app, which also matches what was in Home Assistant with the value of 0.026. If I change the value to yours I also get 0.32043 €/kWh for that time (or I would, I actually computed it using the template tab and [the raw value from ENTSOE](https://transparency.entsoe.eu/transmission-domain/r2/dayAheadPrices/show?name=&defaultValue=false&viewType=GRAPH&areaType=BZN&atch=false&dateTime.dateTime=17.04.2023+00:00%7CCET%7CDAY&biddingZone.values=CTY%7C10YNL----------L!BZN%7C10YNL----------L&resolution.values=PT15M&resolution.values=PT30M&resolution.values=PT60M&dateTime.timezone=CET_CEST&dateTime.timezone_input=CET+(UTC+1)+/+CEST+(UTC+2))).

> when I change the script and save it, the sensors won't ever change

About sensors not changing, I think the current price only changes on the hour, and the average on the day, the exact time depending on [the calculation method](https://github.com/JaccoR/hass-entso-e#calculation-method). The value of the day average might also only match the Vrijopnaam.app value if it's on `rotation`.

**Reply by cyborghound** [reply]
Very strange... this is from my contract for april: 
<img width="562" alt="image" src="https://user-images.githubusercontent.com/25155117/232588341-c8cbdc4d-3f41-43bf-845c-d8cf6ba27d5f.png">
I changed the script back to the one above, and will wait for the change to happen. I will let you know what happens, thanks for the help!

**Reply by GeoffreyFrogeye** [reply]
Actually you made me realize that something is wrong at Vrijopnaam. The value for Inkoopvergoeding shown in the rate sheets I have is also 0.0165 €/kWh, but the calculations on the app are made with 0.026 €/kWh! The former value is what is shown on the rate sheets from February and before. I suppose I didn't notice earlier because when I wanted to update the script in March I saw the calculation made by this plugin still matched the calculation in the app so I only changed the "this should work 'til X" in the header, and today I only checked that the Inkoopvergoeding value was the same since March. I double-checked all that and sent a mail to their customer service explaining the discrepancy, let's see what they have to say about it!

**Reply by cyborghound** [reply]
Just found out that the prices are correct now since the rotation, thanks! I'm still curious if Vrijopnaam does the wrong calculation or if there is some hidden fee, because 0,026 is not 0,0165. But for now I have the correct prices.

**Reply by GeoffreyFrogeye** [reply]
Following my email to them, Vrijopnaam informed me today that the Inkoopvergoeding value they used in the app had indeed not been updated, and it has now been fixed. So while my template showed the prices that were displayed in the app, it didn't have the correct prices either! I now updated the template too (also simplified it).

And now I have to live with the knowledge that my Energy dashboard shows slightly incorrect energy prices for the last few months 😞

**Reply by cyborghound** [reply]
Thanks for the update! That's why my values were not correct this week. I corrected the values yesterday and they seem to be good now. Thanks for the help! And for the energy dashboard, can't you correct that in some way?

**Reply by GeoffreyFrogeye** [reply]
I probably can, but then I have to spend some time to fix it. Neither situation is ideal 😅

**Comment by LeslytH** [comment]
Just as a heads up. I can still confirm the current template works as intended :)
